Scripts/libs/math.lua

```diff
@@ -14,10 +14,38 @@ function math.roundby( x, by)
 
 end
 
+-- parses user-provided number strings
+-- - removes all spaces
+-- - parses binary numbers with the "0b", "0B" prefixes or "b" suffix
+-- - parses hexadecimal numbers with the "0x", "0X" prefixes
+-- returns nil for incorrectly formatted strings
+--
+-- Examples:
+-- math.parsestring(" 1 000    000 ") == 1000000
+-- math.parsestring("0xFF") == 255
+-- math.parsestring("0b1000") == 8
+-- math.parsestring("1000b") == 8
+-- math.parsestring("test") == nil
+function math.parsenumber(input)
+	normalized = string.gsub(input, "%s+", "") -- remove spaces:
+
+	prefix = string.sub(normalized, 1, 2)
+	suffix = string.sub(normalized, -1, -1)
+	if suffix == "b" or suffix == "B" then
+		return tonumber(string.sub(normalized, 1, -2), 2)
+	elseif prefix == "0b" or prefix == "0B" then
+		return tonumber(string.sub(normalized, 3, -1), 2)
+	elseif prefix == "0x" or prefix == "0X" then
+		return tonumber(string.sub(normalized, 3, -1), 16)
+	else
+		return tonumber(normalized, 10)
+	end
+end
+
```

Comment:
I think it's best to move this function outside of the `math` namespace, as it's shared with all mod/vanilla scripts. I know we've done it with more functions, but this could interfere with other mods that define a function with the same name.

Reply:
Right, I've moved `math.parsenumber` to a new lib `input` (maybe there is a clearly better place to put it?) as `mp_parseNumber`